app/javascript/dashboard/helper/voiceRealtime.js

```diff
@@ -0,0 +1,108 @@
+import { CONTENT_TYPES } from 'dashboard/components-next/message/constants';
+
+const VOICE_CALL_TYPES = [CONTENT_TYPES.VOICE_CALL, 12];
+
+const isVoiceCallMessage = data =>
+  VOICE_CALL_TYPES.includes(data?.content_type);
+
+const buildIncomingCallPayload = (data, inboxFromStore) => {
+  const contentData = data.content_attributes?.data || {};
+  return {
+    callSid: contentData.call_sid,
+    conversationId: data.conversation_id,
+    inboxId: data.inbox_id,
+    inboxName: data.meta?.inbox?.name || inboxFromStore?.name,
+    contactName: data.meta?.sender?.name || 'Unknown caller',
+    phoneNumber: data.meta?.sender?.phone_number,
+    isOutbound: contentData.call_direction === 'outbound',
+    callDirection: contentData.call_direction,
+  };
+};
+
+export const handleVoiceMessageUpdated = (app, data) => {
```

Comment:
There are lots of duplicate code between the methods `handleVoiceMessageUpdated` and `handleVoiceMessageCreated`.